src/printer.rs

```diff
@@ -73,27 +99,41 @@ impl<'a> Printer<'a> {
                 );
             }
             SerializationFormat::Text => {
-                if self.log_level >= &LogLevel::Default {
-                    if diagnostics.fixed > 0 {
-                        println!(
-                            "Found {} error(s) ({} fixed).",
-                            diagnostics.messages.len(),
-                            diagnostics.fixed,
-                        );
-                    } else if !diagnostics.messages.is_empty() {
-                        println!("Found {} error(s).", diagnostics.messages.len());
-                    }
-                }
+                self.pre_text(diagnostics);
 
                 for message in &diagnostics.messages {
                     println!("{message}");
                 }
 
-                if self.log_level >= &LogLevel::Default {
-                    if num_fixable > 0 {
-                        println!("{num_fixable} potentially fixable with the --fix option.");
+                self.post_test(num_fixable);
+            }
+            SerializationFormat::Grouped => {
+                self.pre_text(diagnostics);
+
+                let mut filename = String::new();
+
+                for message in &diagnostics.messages {
```

Comment:
I think this assumes that the messages are sorted by filename. Maybe slightly preferable to create a HashMap from filename to messages? That can also help us avoid the clones.

Reply:
Ehh... maybe this is simpler, actually.

Reply:
I took a stab at a hashmap for the sake of it. I can't really make an educated guess on perf, but I think either is simple enough to understand. 🤷 

```rust
let mut map = std::collections::HashMap::new();

for message in &diagnostics.messages {
    map.entry(&message.filename)
        .or_insert_with(Vec::new)
        .push(message);
}

for (filename, messages) in map {
    println!(
        "\n{}:",
        relativize_path(Path::new(&filename)).bold().underline()
    );
    for message in messages {
        println!(
            "    {}{}{} {}  {}",
            message.location.row(),
            ":".cyan(),
            message.location.column(),
            message.kind.code().as_ref().red().bold(),
            message.kind.body(),
        );
    }
}
```